cocos/core/game.ts

```diff
@@ -477,22 +474,24 @@ export class Game extends EventTarget {
      * @en Restart game.
      * @zh 重新开始游戏
      */
-    public restart () {
-        legacyCC.director.once(legacyCC.Director.EVENT_AFTER_DRAW, () => {
+    public restart (): Promise<void> {
+        const afterDrawPromise = new Promise<void>((resolve) =>
+            legacyCC.director.once(legacyCC.Director.EVENT_AFTER_DRAW, () => resolve()));
+        return afterDrawPromise.then(() => {
             // tslint:disable-next-line: forin
-            for (const id in legacyCC.game._persistRootNodes) {
-                legacyCC.game.removePersistRootNode(legacyCC.game._persistRootNodes[id]);
+            for (const id in this._persistRootNodes) {
+                this.removePersistRootNode(this._persistRootNodes[id]);
             }
 
             // Clear scene
             legacyCC.director.getScene().destroy();
             legacyCC.Object._deferredDestroy();
 
             legacyCC.director.reset();
-            legacyCC.game.pause();
-            legacyCC.game._loadRenderPipeline(() => {
-                legacyCC.game.resume();
-                legacyCC.game._safeEmit(legacyCC.Game.EVENT_RESTART);
+            this.pause();
+            return this._setupRenderPipelineAndShowSplashScreen().then(() => {
```

Comment:
@caryliu1999  The `_setupRenderPipelineAndShowSplashScreen` is as its name. You could suggest it with a new name :)

Reply:
Try a shorter name _setRenderPipelineNShowSplash